### PR TITLE
Module 2 Capstone Project

### PR DESCRIPTION
**Pull request by Donmark2k**
### Highlights 🥇

**Mnet Movie project**

**Learning objectives**

- Use webpack to bundle JavaScript.
- Use callbacks and promises.
- Learn how to use proper ES6 syntax.
- Use npm as software packages system.
- Use callbacks and promises.
- Use ES6 modules to write modular JavaScript.
- Send and receive data from an API.
- Use API documentation.
- Understand and use JSON.
- Make JavaScript code asynchronous.

**What I Have Done:**

- Added linters. The [JavaScript + HTML + CSS linter](https://github.com/microverseinc/linters-config/tree/master/html-css-js).
- I create a git flow.
- I create a webpack
- I used Es6 import.
- I send and receive data from an API.
- I make use of JavaScript code asynchronous.
- I add the count function

**Followed Rules:**

- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
- Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
- Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### CODE REVIEW

File: src/modules/addReserve.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        date_start: sDate,
+        date_end: eDate,
+
+      }),
+      headers: { 'content-type': 'application/json' },
+    })
+    .then((resc) => resc.text());
+};
+
+export default addReserve;
```
Comment (OyePriscilla) [inline src/modules/addReserve.js:18]: - [ ] Please kindly implement **async** and **await** with the fetch method to send and receive data from the API. The **async** and **await** keywords enable asynchronous, promise-based behavior to be written in a cleaner style. ℹ️ 

Kindly implement this for **addComment.js**, **fetchApiRes.js** and **fetchapi.js** modules as well (you may want to refactor your code easily as we have it in **addlikes.js**).

## Project requirement
DSC_7491

File: src/template.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          </li>
+          <li>
+            <a href="">new movies</a>
+          </li>
+          <li>
+            <a href="">old movies</a>
+          </li>
+        </ul>
+      </nav>
+    </header>
```
Comment (OyePriscilla) [inline src/template.html:29]: - [ ] Can you please work more a bit on the typography of the header contents? The contrast between the text color and the background-color is too close, and I think the font-size supposed to be increased as well. 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Donmark2k, @AbdimulhinYussuf3675, @Tobby8629 👍, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

so-close-so-close-seth-meyers

## Suggested changes

- [ ] Please kindly remove all these templates from your README file in order to have a professional README file. 👍 
DSC_7491



_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 3]
```diff
@@ -60,25 +60,30 @@ After you're finished please remove all the comments and instructions!
...
 
 ## 🛠 Built With <a name="built-with"></a>
+- HTML
+- CSS
+- Javascript
 
 ### Tech Stack <a name="tech-stack"></a>
 
-> Describe the tech stack and include only the relevant sections that apply to your project.
+> Describe the tech stack and include only the relevant sections that apply to your project. -->
```
Comment (OyePriscilla) [inline README.md:75]: - [ ] Please kindly remove all these templates from your README file in order to have a professional README file. 👍

File: src/modules/fetchApiRes.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/cXyK0mK0Ha1BkQqOi8Kz/reservations?item_id=${id}`)
+    .then((res) => res.json())
+    .then((each) => {
+      const comsec = document.querySelector('.test');
+      arr.push(...arr, ...each);
+      each.map((e) => {
+        resLayout(e, comsec);
+        const header = document.querySelector('.head');
+        header.innerHTML = `${each.length}`;
+        return e;
```
Comment (osamaashraf6) [inline src/modules/fetchApiRes.js:14]: - [ ] As stated 😃 by the previous reviewer kindly 😊 try to implement async and await with the fetch method to send and receive data from the API. The async and await keywords enable asynchronous, promise-based behavior to be written in a cleaner style.💯

File: src/modules/addcomment.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    {
+      method: 'post',
+      body: JSON.stringify({
+        item_id: dataid,
+        username: name,
+        comment: text,
+      }),
+      headers: { 'content-type': 'application/json' },
+    })
+    .then((resc) => resc.text());
```
Comment (osamaashraf6) [inline src/modules/addcomment.js:12]: - [ ] As stated 😃 by the previous reviewer kindly 😊 try to implement async and await with the fetch method to send and receive data from the API. The async and await keywords enable asynchronous, promise-based behavior to be written in a cleaner style.💯

File: src/modules/addReserve.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+        item_id: dataid,
+        username: name,
+        date_start: sDate,
+        date_end: eDate,
+
+      }),
+      headers: { 'content-type': 'application/json' },
+    })
+    .then((resc) => resc.text());
```
Comment (osamaashraf6) [inline src/modules/addReserve.js:15]: - [ ] As stated 😃 by the previous reviewer kindly 😊 try to implement async and await with the fetch method to send and receive data from the API. The async and await keywords enable asynchronous, promise-based behavior to be written in a cleaner style.💯

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi @Donmark2k 😊, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
you-are-near


## Highlights💯🎈🪄✔️

- [x] Descriptive READme file in a professional way.✔️
- [x] No linter errors. ✔️
- [x] Good usage in correcting GitHub flow.✔️
- [x] Nice design.✔️


## Suggested changes


- Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle:  🎉  🎉  🎉 :clap:  :clap: 
Hi @AbdimulhinYussuf3675 & @Tobby8629

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  🎉 🎉 🎉 



## Highlights :heavy_check_mark: 
- [ ] No linter errors :white_check_mark: 
- [ ] Repository has a professional readme file :white_check_mark: 
- [ ] Pull request has a descriptive title and summary :white_check_mark: 
- [ ] Implemented required changes from previous reviewer :white_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tobby8629** [comment]
Thank you 🙏🏾 !!!!

**Comment by OyePriscilla** [comment]
@Donmark2k, @Tobby8629, and @AbdimulhinYussuf3675; Please kindly take note that you should have defined the involvement API once and import it to where it's needful. This will make your codebase DRY according to the best practices. ℹ️ 

**For Example**
**> const baseApi = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/'.**

I'm so sorry for the late observation. 🙏

**Comment by AbdimulhinYussuf3675** [comment]
@OyePriscilla, Sure we will work on that. Thanks

**Pull request merged**